configure.ac

```diff
@@ -546,7 +546,28 @@ esac
 
 AC_ARG_WITH(iodbc,
 AS_HELP_STRING([--with-iodbc=DIR], [build odbc driver against iODBC in DIR]))
-if test "x$with_iodbc" != "x" -a "x$with_iodbc" != "xno"; then
+
+if test "x$with_iodbc" = "xyes"; then
```

Comment:
Didn't check but maybe we would need to change documentation too to state that directory is optional.